gh release download: add specific release ID parameter

v1v opened the pull request:
### What

Download assets from a specific release ID

### Why

This approach will allow to download any assets from draft releases too in addition to  formal releases.

### Use case

Given a draft release I'd like to download all the assets to run some validations before transitioning the release from draft to ready

https://stackoverflow.com/questions/60792587/how-do-i-download-a-file-from-a-github-draft is the reference I've found about doing something similar.


### Examples

```
Download all the artifacts for a specific release-id
$ gh release download --release-id 63259499
```

### Issues

Fixes https://github.com/cli/cli/issues/5443

cliAutomation commented:
Hi! Thanks for the pull request. Please ensure that this change is linked to an issue by mentioning an issue number in the description of the pull request. If this pull request would close the issue, please put the word 'Fixes' before the issue number somewhere in the pull request body. If this is a tiny change like fixing a typo, feel free to ignore this message.

mislav commented:
> Download all the artifacts for a specific release-id
> `$ gh release download --release-id 63259499`

1. How would the user obtain the release ID in the first place?
2. It's already possible to download assets from a draft release that has a tag name: just use the tag name. `gh release download <tag-name>`

v1v committed:
add release-id argument to the release download command

This is just the argument, implementation details to be done in the next commits

v1v committed:
Add release-id implementation

v1v commented:
Thanks Mislav

> How would the user obtain the release ID in the first place?

Good question, I'd rather prefer to delegate that functionality to somewhere else, for instance the `gh api` command can provide that functionality

```bash
$ gh api repos/{owner}/{repo}/releases \
   --jq '.[] | {(.name): .id, draft: .draft, tag: .tag_name}'
```

> It's already possible to download assets from a draft release that has a tag name: just use the tag name. gh release download <tag-name>

As far as I see ` gh release download <tag-name>` does not support multiple releases for the same `tag-name`. This might not be the ideal case, but given the below output from the above command:

```
{"draft":true,"tag":"v1.5","v1.5-test":63259499}
{"draft":false,"tag":"v1.5","v1.5":63139192}
```

If I run `gh release download v1.5`, it will fetch the artifacts from the release `v1.5`.

If I want to download `v1.5-test`, there is no way to download its artifacts but the ones for `v1.5` only.

With this proposal, it's now possible to download assets by release id regardless if the same tag is used in different releases,

For instance, the below commands are the ones to search for the release with name `v1.5-test` to get the release-id, and then download its assets

```bash
## Create temporary folder and validate no artifacts were previously downloaded
$ mkdir test && ls -l | wc -l
       1

## Given the release name v1.5-test
$ RELEASE_NAME=v1.5-test

## And search for the release with name matching RELEASE_NAME and download all the assets
$ gh release download \
   --release-id \
  $(gh api repos/{owner}/{repo}/releases --jq ".[] | select(.name == \"${RELEASE_NAME}\") | .id")

## Then there are downloaded artifacts
$ ls -l | wc -l
     10
```

v1v committed:
Ensure latest release if no release-id

v1v committed:
Add more UTs

mislav commented:
> As far as I see ` gh release download <tag-name>` does not support multiple releases for the same `tag-name`.

You are right: if the draft doesn't have a tag name, or if there are multiple drafts with the same tag name, then you can't select those drafts using the current `gh release <cmd> <tag>` approach.

> > How would the user obtain the release ID in the first place?
> 
> Good question, I'd rather prefer to delegate that functionality to somewhere else, for instance the `gh api` command can provide that functionality

If you are already using `gh api` to fetch information about draft releases, then you can use the same API to fetch assets for a single draft release of your choosing. You do not have to use `gh release download`.

Thank you for explaining the problem and your use case, but I don't yet see a large value in adding this feature when:
1. The feature you are proposing is only useful for selecting draft releases that either don't have a tag name or have a duplicate tag name. All other releases are already selectable via their unique tag name;
2. The feature relies on the user specifying release database IDs, which aren't available to the user anywhere;
3. In order to obtain the IDs, the user would have to access the API directly, and if they're already accessing the API directly, they can just fetch the necessary information themselves.
